pkg/apis/core/validation/validation.go

```diff
@@ -6595,6 +6597,9 @@ func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstrai
 			allErrs = append(allErrs, err)
 		}
 		allErrs = append(allErrs, validateMatchLabelKeys(subFldPath.Child("matchLabelKeys"), constraint.MatchLabelKeys, constraint.LabelSelector)...)
+		if !opts.AllowInvalidTopologySpreadConstraintLabelSelector {
+			allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(constraint.LabelSelector, subFldPath.Child("labelSelector"))...)
```

Comment:
hah, I did not aware there is already a function existed helps to validate the label selector, good to know! 👍